src/main/java/pl/team/carrent/promotion/PromotionRepository.java

```diff
@@ -28,11 +29,13 @@
 
     List<Promotion> findByCars(Car car);
 
-   @Query(value = "select distinct p.* from promotion p" +
-           " join promotion_cars pc on pc.promotion_id=p.id" +
-           " join car c on c.id=pc.cars_id " +
-           " join car_model cm on cm.id = c.car_model_id" +
-           " where UPPER(cm.mark) = UPPER(:mark)", nativeQuery = true)
+    List<Promotion> findByClients(Client client);
+
+    @Query(value = "select distinct p.* from promotion p" +
+            " join promotion_cars pc on pc.promotion_id=p.id" +
+            " join car c on c.id=pc.cars_id " +
+            " join car_model cm on cm.id = c.car_model_id" +
+            " where UPPER(cm.mark) = UPPER(:mark)", nativeQuery = true)
```

Comment:
może spróbuj napisać view który się tworzy w pliku data.sql. podglądnij jak ja to zrobiłem. będzie mniej kodu klejonego tutaj